### DISCUSSION TITLE
using server action mutation from a client component triggers loading page

### DISCUSSION THREAD

**Discussion by marco-digennaro** [vercel/next.js#51396] [App Router]
Basically this is my code:
`
```
  const getActivitiesData = useCallback(async () => {
    const { data } = await getActivities({ id: "1", sort: "title", filter: filteredActivities });
    startTransition(() => setActivities(data));
  }, [filteredActivities]);

  useEffect(() => {
    getActivitiesData();
  }, [getActivitiesData]);
```
`

where _filteredActivities_ is a state array that changes and _activities_ from _setActivities(data)_ is the data array that returns from the fetch and that i loop over.

I expect it would rerender only the portion in the loop, but instead the entire content reloads, i see the content of the _loading.ts_ file

**Comment by michaelxuwu** [comment]
Getting a similar issue, would be good to know what the correct pattern is with regards to Server Mutations without causing the loading UI (`/loading.js`) to display
